utils/file.c

```diff
@@ -77,27 +77,32 @@ int myst_load_file(const char* path, void** data_out, size_t* size_out)
     return ret;
 }
 
-ssize_t myst_writen(int fd, const void* data, size_t size)
+int myst_write_file_fd(int fd, const void* data, size_t size)
 {
-    ssize_t ret = 0;
+    int ret = 0;
     const uint8_t* p = (const uint8_t*)data;
     size_t r = size;
+    ssize_t n;
+
+    if (fd < 0 || !data)
+        ERAISE(-EINVAL);
 
     while (r > 0)
     {
-        ssize_t n = write(fd, p, r);
-
-        if (n == 0)
+        if ((n = write(fd, p, r)) == 0)
             break;
-        else if (n < 0)
-            ERAISE(n);
+
+        if (n < 0)
+            ERAISE((int)-n);
 
         p += n;
         r -= (size_t)n;
     }
 
-done:
+    if (r != 0)
+        ERAISE(-EIO);
```

Comment:
Generally we don't change the error code in the done section.

Reply:
This is above the done section.